### ISSUE TITLE
Uncaught ReferenceError: process is not defined

### ISSUE THREAD

**Issue by wesleyt95** [EasyPost/easypost-node#229]
I installed the package today hoping to test but just the lines: ``` const EasyPost = require('@easypost/api')``` &
    ```const api = new EasyPost(API_Key)``` causes errors: ```Uncaught ReferenceError: process is not defined
    at Module../easypost.js (easypost.js:36:1)```  

easypost.js: ```export const UA_INFO = {
  client_version: pkg.version,
  lang: 'nodejs',
  lang_version: process.version,
  publisher: 'easypost',
  platform: os.platform(),
};```

**Comment by wesleyt95** [comment]
Edit: ```const api = new EasyPost(API_Key)``` isn't causing the error, so ```const EasyPost = require('@easypost/api')``` must be the problem... Did I do something wrong?

**Comment by wesleyt95** [comment]
Uninstalled and reinstalled everything... new error: ```Compiled with problems:X

ERROR in ./node_modules/@easypost/api/easypost.js 3870:21-34

Module not found: Error: Can't resolve 'os' in 'C:\Users\Wes\YSC\ysctechclub\node_modules\@easypost\api'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "os": require.resolve("os-browserify/browser") }'
	- install 'os-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "os": false }```

**Comment by Justintime50** [comment]
Can you provide some additional details to help us determine what's going on here? What version of Node are you using? What kind of app are you trying to build and how are you including the EasyPost library (are you using webpack, are you simply including it in a node app, are you needing to use a polyfill based on the version of Node you are using)? If you are using anything older than Node 12, please read the README for setup instructions.

**Comment by wesleyt95** [comment]
@Justintime50 

Node: v16.6.1
Webpack v5.71
@easypost/api v4.0.0

I'm building my first basic e-commerce React app so I'm not sure what I'm doing wrong all I did was type ```const EasyPost = require('@easypost/api')``` at the top of the file; according to my version's I don't think I should need a 'polyfill'. I believe I fixed the 'os' error  by fixing some wrong versions in my package.json file.

I also referenced [this](https://stackoverflow.com/questions/41359504/webpack-bundle-js-uncaught-referenceerror-process-is-not-defined) but it lead to more problems

**Comment by wesleyt95** [comment]
also in my package.json:
```
  "devDependencies": {
    "process": "^0.11.10"
  }
```

**Comment by Justintime50** [comment]
This package is intended to be used as a backend and not included in frontend code as it's written in Node and not Javascript. What I'd suggest is you create a backend (API) that exposes endpoints to do things such as create a shipment and purchase a label. Then you can have your React app (frontend) hit those endpoints and send the data needed to perform those actions. There are various packages that are only available in Node that are not available via a browser (os, process, etc) which is why you are receiving those errors. You'll either need to:

1. Use Webpack 4 as the error message stated it included those pollyfills for you by default
2. Use pollyfills in Webpack 5 to get the same result (`process` has a browser shim that can be used)
3. Separate your frontend and backend and only install the EasyPost Node package on the backend (recommended)

This is more a problem with your Webpack setup and Node vs Javascript than it is with this library. Using one of the above three approaches should get you on your way. Best of luck!

**Comment by wesleyt95** [comment]
Thanks for responding, I did the following commands after reading about polyfills and webpack 5... I thought this would do the trick but process is still undefined; I don't want to give up on this too soon do you have any critique for the following?


Install patch-package: ```yarn add patch-package```

Modify ```webpack.config.js```:
```
module.exports = {
  //...
  resolve: {
    fallback: {
      os: require.resolve('os-browserify/browser'),
      path: require.resolve('path-browserify'),
      process: require.resolve('process/browser'),
    },
  },
};
```

Run ```yarn patch-package react-scripts```

Add a postinstall script to package.json: "postinstall": "yarn patch-package":

```
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject",
    "postinstall": "yarn patch-package"
  },
```

**Comment by wesleyt95** [comment]
@Justintime50 Okay I did a little research and chose to implement firebase cloud functions and connected it to my front end... I think it's partially working! I checked the logs from firebase functions I received a tracking link, a rate, shipping label image link and a carrier_account_id; does that mean it's working properly? There's a lot of info being returned in the firebase logs though and I'm not sure how to access it within my app if that makes sense? the localhost:3000 console.log is returning null for some reason, I asked the same question on stackoverflow so my code is [here](https://stackoverflow.com/questions/71812612/easypost-node-js-example-difficulties)

Thanks!
